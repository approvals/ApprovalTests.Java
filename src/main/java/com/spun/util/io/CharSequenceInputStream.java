package com.spun.util.io;

import java.io.IOException;
import java.io.InputStream;

public class CharSequenceInputStream extends InputStream {
	@Override
	public int read() throws IOException {
		// TODO Auto-generated method stub
		return 0;
	}
}
