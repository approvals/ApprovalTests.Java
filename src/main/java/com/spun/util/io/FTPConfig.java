package com.spun.util.io;


/**
  * A static class of convence functions for Files
  **/
public class FTPConfig
{
  public static final int DEFUALT_FTP_PORT = 21;
  public String host, userName, password;
  public int port = 0;
  /***********************************************************************/
  public FTPConfig(String host,int port, String userName,String password)
  {
    this.host = host;
    this.port = port;
    this.userName = userName;
    this.password = password;
  }
  /************************************************************************/
  /************************************************************************/
}