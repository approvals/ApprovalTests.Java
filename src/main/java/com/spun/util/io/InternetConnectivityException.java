package com.spun.util.io;

import java.net.InetAddress;

import org.apache.commons.net.EchoTCPClient;

import com.spun.util.MySystem;

/**
 * A static class of convenience functions for Files
 **/
public class InternetConnectivityException extends Error
{
  private String htmlText;
  private Throwable cause;

  /***********************************************************************/
  public void setCause(Throwable t)
  {
    this.cause = t;
  }
  /***********************************************************************/
  
  public InternetConnectivityException(String string)
  {
    super(string);
  }
  /***********************************************************************/
  public void setHTMLText(String htmlText)
  {
    this.htmlText = htmlText;
  }
  /***********************************************************************/
  public String getHTMLText()
  {
    return htmlText;
  }
  /***********************************************************************/
  public static InternetConnectivityException testInternetConnectivity()
  {
    String[] sites = {"www.google.com", "www.yahoo.com", "www.msn.com"};
    
    for (int i = 0; i < sites.length; i++)
    {
      if (pingSite(sites[i])) { return null; }
    }
    return new InternetConnectivityException("There is no internet connection.");
  }

  /***********************************************************************/
  public String toString()
  {
    return super.toString() + "\n" + (cause == null ? "" : "Causation:  " + cause.toString());
  }
  /***********************************************************************/

  private static boolean pingSite(String site)
  {
    try
    {
      EchoTCPClient client = new EchoTCPClient();
      client.setDefaultTimeout(1000);
      client.connect(InetAddress.getByName(site), 80);
      client.disconnect();
      return true;
    }
    catch (Exception e)
    {
      MySystem.warning(site, e);
      return false;
    }

  }
  /***********************************************************************/
  /************************************************************************/

}