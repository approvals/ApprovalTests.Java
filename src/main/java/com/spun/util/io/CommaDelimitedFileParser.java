package com.spun.util.io;

import java.io.BufferedReader;
import java.io.File;
import java.io.IOException;
import java.io.Reader;
import java.io.StringReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import com.spun.util.StringUtils;

public class CommaDelimitedFileParser
{
  /***********************************************************************/
  public static String[][] parse(File databaseFile)
  {
    return parse(FileUtils.readFileWithSuppressedExceptions(databaseFile));
  }
  /***********************************************************************/
  public static String[][] parse(String data)
  {
    return parse(new StringReader(data));
  }
  /***********************************************************************/
  public static String[][] parse(Reader data)
  {
    try
    {
      List<String[]> records = new ArrayList<String[]>();
      BufferedReader reader = new BufferedReader(data);
      boolean done = false;
      StringBuffer pastLines = null;
      while (!done)
      {
        String line = reader.readLine();
        if (line == null)
        {
          done = true;
        }
        else
        {
          if (pastLines != null)
          {
            pastLines.append("\r\n");
            pastLines.append(line);
            line = pastLines.toString();
          }
          String[] parseLine = parseLine(line);
          if (parseLine == null)
          {
            if (pastLines == null)
            {
              pastLines = new StringBuffer(line);
            }
          }
          else
          {
            records.add(parseLine);
            pastLines = null;
          }
        }
      }
      return records.toArray(new String[records.size()][]);
    }
    catch (IOException e)
    {
      throw new RuntimeException(e);
    }
  }
  /***********************************************************************/
  private static String[] parseLine(String line) throws IOException
  {
    String[] rawtokens = StringUtils.split(line, ",", false);
    boolean in = false;
    List<String> tokens = new ArrayList<String>();
    String fullToken = null;
    for (int i = 0; i < rawtokens.length; i++)
    {
      String string = rawtokens[i];
      if (in)
      {
        fullToken += ",";
        fullToken += string;
        if (fullToken.endsWith("\""))
        {
          tokens.add(clearQuotes(fullToken));
          in = false;
        }
      }
      else
      {
        if (string.startsWith("\"") && !string.endsWith("\""))
        {
          fullToken = string;
          in = true;
        }
        else
        {
          tokens.add(clearQuotes(string));
        }
      }
    }
    return in ? null : StringUtils.toArray(tokens);
  }
  /***********************************************************************/
  private static String clearQuotes(String string)
  {
    String s = (string.startsWith("\"") && string.endsWith("\""))
        ? string.substring(1, string.length() - 1)
        : string;
    return s.trim();
  }
  /***********************************************************************/
  public static Map<String, String>[] parseToMap(File file)
  {
    try
    {
      return parseToMap(FileUtils.readFileWithSuppressedExceptions(file));
    }
    catch (RuntimeException e)
    {
      System.out.println("Filename : " + file.toString());
      throw e;
    }
  }
  /***********************************************************************/
  public static Map<String, String>[] parseToMap(String data)
  {
    return parseToMap(new StringReader(data));
  }
  public static Map<String, String>[] parseToMap(Reader data)
  {
    String[][] out = parse(data);
    Map<String, String>[] maps = new HashMap[out.length - 1];
    String[] labels = out[0];
    for (int i = 1; i < out.length; i++)
    {
      Map<String, String> map = new HashMap<String, String>(labels.length);
      for (int j = 0; j < labels.length; j++)
      {
        map.put(labels[j], out[i][j]);
      }
      maps[i - 1] = map;
    }
    return maps;
  }
  /***********************************************************************/
  /***********************************************************************/
}