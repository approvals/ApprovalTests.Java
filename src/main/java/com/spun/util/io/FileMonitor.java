package com.spun.util.io;

import java.io.File;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.Timer;
import java.util.TimerTask;

public class FileMonitor
{
  private Timer             timer;
  private Map<File, Long>   files;
  private Set<FileListener> listeners;
  public FileMonitor(long pollingInterval)
  {
    files = new HashMap<File, Long>();
    listeners = new HashSet<FileListener>();
    timer = new Timer(true);
    timer.schedule(new FileMonitorNotifier(), 0, pollingInterval);
  }
  public void stop()
  {
    timer.cancel();
  }
  public void addFile(File file)
  {
    if (!files.containsKey(file))
    {
      files.put(file, (file.exists() ? file.lastModified() : -1L));
    }
  }
  public void removeFile(File file)
  {
    files.remove(file);
  }
  public void addListener(FileListener fileListener)
  {
    listeners.add(fileListener);
  }
  public void removeListener(FileListener fileListener)
  {
    listeners.remove(fileListener);
  }
  private class FileMonitorNotifier extends TimerTask
  {
    public void run()
    {
      for (File file : files.keySet())
      {
        long lastModifiedTime = ((Long) files.get(file)).longValue();
        long newModifiedTime = file.exists() ? file.lastModified() : -1;
        if (newModifiedTime != lastModifiedTime)
        {
          files.put(file, newModifiedTime);
          for (FileListener listener : listeners)
          {
            listener.fileChanged(file);
          }
        }
      }
    }
  }
  public interface FileListener
  {
    void fileChanged(File file);
  }
}
