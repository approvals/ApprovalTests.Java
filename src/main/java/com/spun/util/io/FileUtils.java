package com.spun.util.io;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.DataOutputStream;
import java.io.File;
import java.io.FileFilter;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.Reader;
import java.nio.channels.FileChannel;
import java.util.ArrayList;
import java.util.List;

import com.spun.util.ArrayUtils;
import com.spun.util.Asserts;
import com.spun.util.ObjectUtils;

/**
 * A static class of convenience functions for Files
 **/
public class FileUtils
{
  /***********************************************************************/
  /**
   * @see java.io.File.createTempFile(String,String)
   **/
  public static File createTempDirectory() throws IOException
  {
    File tempFile = File.createTempFile("TEMP", null);
    tempFile.delete();
    tempFile.mkdirs();
    return tempFile;
  }
  /***********************************************************************/
  public static void deleteDirectory(File directory) throws IOException
  {
    // delete all directory
    File directories[] = directory.listFiles(new SimpleDirectoryFilter());
    for (int i = 0; i < directories.length; i++)
    {
      deleteDirectory(directories[i]);
    }
    // Delete all Files.
    File files[] = directory.listFiles(new SimpleFileFilter());
    for (int i = 0; i < files.length; i++)
    {
      files[i].delete();
    }
    // delete self.
    directory.delete();
  }
  public static String readFromClassPath(Class clazz, String string)
  {
    final InputStream resourceAsStream = clazz.getResourceAsStream(string);
    if (resourceAsStream == null)
    {
      String message = String.format("Could not find %s from %s", string, clazz.getName());
      throw new RuntimeException(message);
    }
    String resource = FileUtils.readStream(resourceAsStream);
    return resource;
  }
  /***********************************************************************/
  public static File[] getRecursiveFileList(File directory)
  {
    return getRecursiveFileList(directory, new SimpleFileFilter());
  }
  /***********************************************************************/
  public static File[] getRecursiveFileList(File directory, FileFilter filter)
  {
    ArrayList<File> list = new ArrayList<File>();
    if (!directory.isDirectory()) { throw new Error("File is not a directory: " + directory.getName()); }
    File directories[] = directory.listFiles(new SimpleDirectoryFilter());
    for (int i = 0; i < directories.length; i++)
    {
      ArrayUtils.addArray(list, getRecursiveFileList(directories[i], filter));
    }
    File files[] = directory.listFiles(filter);
    ArrayUtils.addArray(list, files);
    return list.toArray(new File[list.size()]);
  }
  /***********************************************************************/
  public static void copyFile(File in, File out)
  {
    try
    {
      FileChannel inChannel = null, outChannel = null;
      try
      {
        out.getParentFile().mkdirs();
        inChannel = new FileInputStream(in).getChannel();
        outChannel = new FileOutputStream(out).getChannel();
        outChannel.transferFrom(inChannel, 0, inChannel.size());
      }
      finally
      {
        if (inChannel != null)
        {
          inChannel.close();
        }
        if (outChannel != null)
        {
          outChannel.close();
        }
      }
    }
    catch (Exception e)
    {
      ObjectUtils.throwAsError(e);
    }
  }
  public static void copyStream(InputStream in, OutputStream out) throws IOException
  {
    byte[] buf = new byte[1024];
    int i = 0;
    while ((i = in.read(buf)) != -1)
    {
      out.write(buf, 0, i);
    }
    in.close();
    out.close();
  }
  /***********************************************************************/
  public static void redirectInputToFile(String fileName, InputStream in) throws Exception
  {
    FileOutputStream fos = new FileOutputStream(new File(fileName), false);
    copyStream(in, fos);
  }
  /***********************************************************************/
  public static void copyFileToDirectory(String file, File tempDir) throws Exception
  {
    File in = new File(file);
    File out = new File(tempDir, in.getName());
    copyFile(in, out);
  }
  /***********************************************************************/
  public static void writeFile(File file, String text) throws IOException
  {
    Asserts.assertNotNull("Writing to file: " + file, text);
    file.getCanonicalFile().getParentFile().mkdirs();
    BufferedWriter out = new BufferedWriter(new FileWriter(file));
    out.write(text);
    out.close();
  }
  /***********************************************************************/
  public static void writeFileQuietly(File file, String text)
  {
    try
    {
      writeFile(file, text);
    }
    catch (IOException e)
    {
      ObjectUtils.throwAsError(e);
    }
  }
  public static void writeFile(File file, CharSequence data) throws IOException
  {
    Asserts.assertNotNull("Writing to file: " + file, data);
    file.getCanonicalFile().getParentFile().mkdirs();
    DataOutputStream writer = new DataOutputStream(new FileOutputStream(file));
    for (int i = 0; i < data.length(); i++)
    {
      writer.write(data.charAt(i));
    }
    writer.close();
  }
  public static void writeFile(File file, InputStream data) throws IOException
  {
    Asserts.assertNotNull("Writing to file: " + file, data);
    file.getCanonicalFile().getParentFile().mkdirs();
    copyStream(data, new FileOutputStream(file));
  }
  /***********************************************************************/
  public static String readFile(String absolutePath) throws IOException
  {
    return readFile(new File(absolutePath));
  }
  /***********************************************************************/
  public static String readFile(File file) throws IOException
  {
    BufferedReader in = new BufferedReader(new FileReader(file));
    return readBuffer(in);
  }
  public static String readBuffer(BufferedReader in) throws IOException
  {
    StringBuffer string = new StringBuffer();
    while (in.ready())
    {
      string.append(in.readLine());
      string.append("\n");
    }
    in.close();
    return string.toString();
  }
  /************************************************************************/
  public static String readFileWithSuppressedExceptions(File databaseFile)
  {
    try
    {
      return FileUtils.readFile(databaseFile);
    }
    catch (FileNotFoundException e)
    {
      throw new RuntimeException("Invalid file '" + databaseFile.getAbsolutePath() + "'", e);
    }
    catch (IOException e)
    {
      throw new RuntimeException(e);
    }
  }
  /************************************************************************/
  public static File saveToFile(String prefix, Reader input)
  {
    File file;
    BufferedWriter bw = null;
    try
    {
      file = File.createTempFile(prefix, null);
      bw = new BufferedWriter(new FileWriter(file));
      BufferedReader inputReader = new BufferedReader(input);
      String thisLine;
      while ((thisLine = inputReader.readLine()) != null)
      {
        bw.write(thisLine);
        bw.newLine();
      }
      inputReader.close();
    }
    catch (IOException e)
    {
      throw new RuntimeException("Unable to store order: " + e.getMessage(), e);
    }
    finally
    {
      try
      {
        if (bw != null)
        {
          bw.close();
        }
      }
      catch (IOException e)
      {}
    }
    return file;
  }
  /************************************************************************/
  public static String getDirectoryFriendlyName(String name)
  {
    if (name == null) { return ""; }
    StringBuffer result = new StringBuffer();
    for (int i = 0; i < name.length(); i++)
    {
      char c = name.charAt(i);
      switch (c)
      {
        case '.' :
          break;
        default :
          result.append(c);
          break;
      }
    }
    return result.toString();
  }
  /************************************************************************/
  /************************************************************************/
  public static String getExtensionWithDot(String filename)
  {
    int p = filename.lastIndexOf('.');
    return filename.substring(p);
  }
  public static String getExtensionWithoutDot(String filename)
  {
    return getExtensionWithDot(filename).substring(1);
  }
  public static void createIfNeeded(String file) throws IOException
  {
    File f = new File(file);
    if (!f.exists())
    {
      writeFile(f, "");
    }
  }
  public static String readStream(InputStream resourceAsStream)
  {
    BufferedReader reader = new BufferedReader(new InputStreamReader(resourceAsStream));
    String resource = null;
    try
    {
      resource = FileUtils.readBuffer(reader);
    }
    catch (IOException e)
    {
      ObjectUtils.throwAsError(e);
    }
    return resource;
  }
  public static char[] loadResourceFromClasspathAsBytes(Class clazz, String name) throws IOException
  {
    return extractBytes(clazz.getResourceAsStream(name));
  }
  public static char[] extractBytes(final InputStream resourceAsStream) throws IOException
  {
    ArrayList<Character> data = new ArrayList<Character>();
    int b = resourceAsStream.read();
    while (b != -1)
    {
      data.add(new Character((char) b));
      b = resourceAsStream.read();
    }
    return FileUtils.toChars(data);
  }
  public static char[] toChars(List<Character> data)
  {
    char[] out = new char[data.size()];
    for (int i = 0; i < out.length; i++)
    {
      out[i] = data.get(i);
    }
    return out;
  }
  public static boolean isNonEmptyFile(String approved)
  {
    File file = new File(approved);
    return file.exists() && file.length() > 0;
  }
  public static void ensureParentDirectoriesExist(File file)
  {
    File dir = file.getParentFile();
    if (!dir.exists())
    {
      dir.mkdirs();
    }
  }
}