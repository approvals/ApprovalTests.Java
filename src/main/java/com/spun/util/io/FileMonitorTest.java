package com.spun.util.io;

import java.io.File;
import junit.framework.TestCase;
import com.spun.util.io.FileMonitor.FileListener;

public class FileMonitorTest extends TestCase
{
  FileMonitor monitor;
  File        root;
  boolean     done;
  protected void setUp() throws Exception
  {
    File filea = File.createTempFile("tmp", "tmp");
    root = new File(filea.getParent() + File.separator + "fileMonitorRoot");
    root.mkdir();
    filea.delete();
    monitor = new FileMonitor(100);
    monitor.addFile(root);
    monitor.addListener(this.new TestListener());
    Thread.sleep(1000);
  }
  protected void tearDown() throws Exception
  {
    monitor.stop();
    root.delete();
  }
  public void testNewFileDetected() throws Exception
  {
    done = false;
    File targetFile = File.createTempFile("tmp", ".dbf", root);
    for (int i = 0; i < 100 && !done; i++)
    {
      Thread.sleep(1000);
    }
    try
    {
      assertEquals("File processed", true, done);
    }
    finally
    {
      targetFile.delete();
    }
  }
  private class TestListener implements FileListener
  {
    public void fileChanged(File file)
    {
      done = true;
    }
  }
}
