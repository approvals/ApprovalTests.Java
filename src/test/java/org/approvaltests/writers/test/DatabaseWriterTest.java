package org.approvaltests.writers.test;

import java.sql.ResultSet;
import java.sql.ResultSetMetaData;
import java.sql.SQLException;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.easymock.EasyMock;

import com.spun.util.io.CommaDelimitedFileParser;
import com.spun.util.io.FileUtils;

public class DatabaseWriterTest extends TestCase
{
  public void testSimpleQuery() throws Exception
  {
    //ResultSet rs = queryDatagbase();
    ResultSet rs = mockResultSetFromFile("query.csv");
    Approvals.verify(rs);
  }
  private ResultSet queryDatagbase() throws SQLException
  {
    return mockResultSetFromFile("sample_result_set.csv");
  }
  private ResultSet mockResultSetFromFile(String fileName) throws SQLException
  {
    String resultSet = FileUtils.readFromClassPath(getClass(), fileName).trim();
    String[][] data = CommaDelimitedFileParser.parse(resultSet);
    return mockResultSet(data);
  }
  private ResultSet mockResultSet(String[][] data) throws SQLException
  {
    ResultSet rs = EasyMock.createMock(ResultSet.class);
    ResultSetMetaData metaData = EasyMock.createMock(ResultSetMetaData.class);
    EasyMock.expect(rs.getMetaData()).andReturn(metaData).anyTimes();
    EasyMock.expect(metaData.getColumnCount()).andReturn(data[0].length).anyTimes();
    for (int i = 0; i < data[0].length; i++)
    {
      EasyMock.expect(metaData.getColumnName(i + 1)).andReturn(data[0][i]).anyTimes();
    }
    for (int i = 1; i < data.length; i++)
    {
      EasyMock.expect(rs.next()).andReturn(true);
      for (int j = 0; j < data[i].length; j++)
      {
        EasyMock.expect(rs.getString(j + 1)).andReturn(data[i][j]);
      }
    }
    EasyMock.expect(rs.next()).andReturn(false);
    EasyMock.replay(rs);
    EasyMock.replay(metaData);
    return rs;
  }
}
