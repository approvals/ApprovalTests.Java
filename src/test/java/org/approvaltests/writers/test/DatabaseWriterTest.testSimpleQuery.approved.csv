pkey,name,occupation,company_name,email,phone_number,fax_number,web_site,address,status,add_date
"1","Rich Yumul","Software Consultant","Sage Tree Solutions","rmy@sagetree.net","(866)728-9100","","","1775 Sage Tree Ct","Active","2006-12-31 21:00:00.0"
"2","Llewellyn Falco","Agile Programing Coach","Spun Labs Inc.","info@spunlabs.com","(858)336-6560","","http://www.spunlabs.com","1121 Chalcedony Ave, San Diego 92109","Active","2006-12-31 21:00:00.0"
"3","Paul Hart","Billiard Services","","","(858)335-4828","","","","Inactive","2006-12-31 21:00:00.0"
"4","Allen Stovall","Acupuncturist","5 Element Institute","","(888)423-4325","","http://www.harmonyhealthlink.com/","3678 4th Ave. San Diego, CA. 92103","Active","2006-12-31 21:00:00.0"
"5","Deb Davies","Massage Therapist","","shiatsubydeb@yahoo.com","","","","","Inactive","2006-12-31 21:00:00.0"
"6","Shettal Bhor","Dentist","","","","","","","Inactive","2006-12-31 21:00:00.0"
"7","Christina Sanchez","Chiropractor","","choicechiro@gmail.com","","","","","Inactive","2006-12-31 21:00:00.0"
"8","Kim Priebe","Loan Officer","","kim@sdlendingco.com","(619)546-7899","","","","Inactive","2006-12-31 21:00:00.0"
"9","Peter Manosh","Personal Trainer","","pptans@msn.com","","","","","Inactive","2006-12-31 21:00:00.0"
"10","Julia Orpheus","Residential Realtor","","juliaorpheus@yahoo.com","(858)729-6408","","","","Inactive","2006-12-31 21:00:00.0"
"11","Casey Ross","Life Insurance","","cross@ft.newyorklife.com","","","","","Inactive","2006-12-31 21:00:00.0"
"12","Marcella Cota","Insurance Agent","","mdiaz3@farmersagent.com","","","","","Inactive","2006-12-31 21:00:00.0"
"13","Ginger Wilshin","Printing","","gwilshin@rushpress.com","","","","","Inactive","2006-12-31 21:00:00.0"
"14","Carol Lord","Marketing","","carollord@cox.net","(858)829-2624","","","","Active","2006-12-31 21:00:00.0"
"15","Michele Esposito","Organizer","","theorganizedroom@yahoo.com","","","","","Inactive","2006-12-31 21:00:00.0"
"16","Suzanne Steel","Landscaper","","","","","","","Inactive","2006-12-31 21:00:00.0"
"17","Kevin Manaugh","Financial Advisor","","","","","","","Inactive","2006-12-31 21:00:00.0"
"18","Ben Burnette","Zija","","drbenburnette@yahoo.com","(619)243-4301","","","","Inactive","2006-12-31 21:00:00.0"
"19","Ryan Beilharz","Dry Cleaning","","ryan@relaxxdrycleaning.com","","","","","Inactive","2007-02-14 21:00:00.0"
"20","Thomas Mattingly","Appraiser","","customerservice@appraise-tech.com","(619)255-3430","","","","Inactive","2007-02-14 21:00:00.0"
"21","Paul McNamara","Banker","","paul.mcnamara@uboc.com","","","","","Inactive","2006-11-30 21:00:00.0"
"22","Donna Hamilton","Escrow","","donnahamilton@aplusescrow.com","(619)595-1241","","","","Inactive","2007-01-31 21:00:00.0"
"23","Kim-Marie","Letip","","","","","","","InterChapter","2006-11-06 21:00:00.0"
"24","Leslie Hill","Printer","","","","","","","InterChapter","2007-02-21 21:00:00.0"
"25","Warren Miller","Insurance Agent","","","","","","","Inactive","2007-11-06 21:00:00.0"
"26","Eli Littrell","Concierge","","","","","","","Inactive","2007-11-06 21:00:00.0"
"28","Glen Story","Construction","","","","","","","Inactive","2007-11-06 21:00:00.0"
"29","Jesse LeValley","Construction","","","","","","","InterChapter","2007-02-28 21:00:00.0"
"30","Barb Crum","Esthetician","","","","","","","Inactive","2007-03-21 21:00:00.0"
"31","Kirstie Berzanski","SendOutCards","","kb@sendoutcards.com","","","","","Inactive","2007-03-21 21:00:00.0"
"32","Blake Wimple","Payroll","","blake_wimple@adp.com","","","","","Inactive","2007-03-21 21:00:00.0"
"33","Mikel Bruce","Website Design","","mikel@webflexor.com","(619)795-1894","","","","Active","2007-04-18 21:00:00.0"
"34","Robert Colosia","Personal Injury Attorney","","lawyer@colosia.com","(619)231-8340","","","","Active","2007-04-18 21:00:00.0"
"35","Karen Gibson","Skincare","","kg.1@mac.com","(619)733-8094","","","","Active","2007-04-18 21:00:00.0"
"36","Jackleen Holton","Life Coach","","jackleen@innercirclecoaching.com","","","","","InterChapter","2007-02-18 21:00:00.0"
"37","Ron Jones","Accountant","","","","","","","InterChapter","2007-03-14 21:00:00.0"
"38","Jason Coberly","Attorney","","","","","","","InterChapter","2007-04-14 21:00:00.0"
"39","Ronald Cervi","Carpet Cleaner","","","","","","","InterChapter","2007-04-24 21:00:00.0"
"40","Bart Blakely","Electrition","","blakelydbart@aol.com","","","","","Inactive","2007-05-16 21:00:00.0"
"41","Sumaku Golston","Graphic Artist","","sumaku@gmail.com","(619)944-3334","","","","Inactive","2007-05-16 21:00:00.0"
"42","Felisa Schlosser","Loan Officer","","","","","","","InterChapter","2006-12-31 21:00:00.0"
"43","Andre","Printer","","","","","","","InterChapter","2007-05-20 21:00:00.0"
"44","Rebecca T","Personal Trainer","","","","","","","InterChapter","2007-05-30 21:00:00.0"
"45","Ronald Glasgow II","Financial Advisor","","ronald.glasgow@edwardjones.com","","","","","Inactive","2007-06-13 21:00:00.0"
"46","Kristine Kyllander","Estate Planner","","kc@kyllanderlaw.com","","","","","Inactive","2007-06-13 21:00:00.0"
"47","Jeff Tarlton","Autobody","","jeff@bumperdoc.com","","","","","Inactive","2007-07-04 21:00:00.0"
"48","Craig D'Agnese","Photo","","","","","","","InterChapter","2007-07-04 21:00:00.0"
"49","Tim Chesser","Printer","","","","","","","InterChapter","2007-07-04 21:00:00.0"
"50","Brain Lacer","Massage","","","","","","","InterChapter","2007-07-18 21:00:00.0"
"51","Susan Heberer","Enbroider","","","","","","","InterChapter","2007-07-18 21:00:00.0"
"52","Susan Crowder","Banker","","sue.crowder@uboc.com","(619)221-2834","","","","Active","2007-07-18 21:00:00.0"
"54","Rosemarie","Unknown","","","","","","","InterChapter","2007-07-18 21:00:00.0"
"55","Alex Rohrke","Insurance","","","","","","","InterChapter","2007-07-18 21:00:00.0"
"56","Larry","Accounting","","","","","","","InterChapter","2007-07-11 21:00:00.0"
"57","Andrea Barnhart","eyes","","","","","","","InterChapter","2007-08-01 21:00:00.0"
"58","Rosemarie Cohenour","Hair Stylist","","","(619)574-0172","","","","Inactive","2007-08-29 21:00:00.0"
"59","Matt Walker","Internet Marketing","","matt@bestrank.com","(858)427-0825","","","","Inactive","2007-08-01 21:00:00.0"
"60","Jasmine Guerrero","P&C Insurance","","jguerrero1@farmersagent.com","(619)358-5383","","","","Active","2007-08-29 21:00:00.0"
"61","Drew Lyman","Bankruptcy Attorney","","dlyman@lymanpotente.com","(619)299-2399","","","","Inactive","2007-08-29 21:00:00.0"
"62","Peter Potente","Family Law","","ppotente@lymanpotente.com","(619)299-2399","","","","Inactive","2007-08-29 21:00:00.0"
"63","Johana Finberg","Commerical Insurance","","","","","","","InterChapter","2007-08-29 21:00:00.0"
"64","Erik Pfeffer","Computer Repair","","","","","","","InterChapter","2007-09-05 21:00:00.0"
"66","Margo Parras","unknown","","","","","","","InterChapter","2007-09-19 21:00:00.0"
"67","Dan Parks","Financial Advisor","","","(619)435-7712","","","","Inactive","2007-09-19 21:00:00.0"
"68","Lawrence Bakur","Chiropractor","","","(619)955-3303","","","","Inactive","2007-09-19 21:00:00.0"
"69","Patrick Siefner","painter","","","","","","","InterChapter","2007-10-03 21:00:00.0"
"70","Molly Cooke","Massage","","molly@mia-belle.com","(619)540-9709","","","","Inactive","2007-10-31 21:00:00.0"
"71","Joshua Jones","Life/Health Insurance","","joshua.jones@nmfn.com","(619)234-3111","","","","Inactive","2007-11-14 21:00:00.0"
"72","Aubrey King","CPA","","","","","","","Inactive","2007-12-31 21:00:00.0"
"73","Scott Elliot","Personal Trainer","","","","","","","InterChapter","2007-12-11 21:00:00.0"
"76","Chip Sexton","Massage Therapist","","chipsexton@gmail.com","(619)540-9709","","","","Inactive","2007-12-31 21:00:00.0"
"77","Barbara Lukehart","Send Out Cards","Send Out Cards","Barbara@barbaralukehart.com","(619)928-1200","","","","Active","2008-02-05 21:00:00.0"
"79","Debbie Bonham","Seamstress","","","(619)460-5766","","","","Active","2008-02-09 21:00:00.0"
"80","Dave McKinnon","Personal Trainer","Solid State Kettlebell","","","","","","Inactive","2008-02-22 21:00:00.0"
"81","Eco-Dry","Carpet Cleaner","","","","","","","InterChapter","2008-03-05 21:00:00.0"
"82","Dan Niles","Aeration","","","","","","","Inactive","2008-03-30 21:00:00.0"
"83","Pattty Sterling","Estate Planning","Pacific Coast Trust","pattys@pacificcoasttrust.com","(619)339-7893","","","","Active","2008-04-05 21:00:00.0"
"84","Tom Hamlin","Computer Services","","tom@ltg-technology.com","(619)501-1713","","","","Inactive","2008-04-20 21:00:00.0"
"85","Pete Dvorak","Printer","","","","","","","Inactive","2008-04-20 21:00:00.0"
"86","Brent Morton","Carpet Cleaning","","preferredshakay@yahoo.com","(619)283-6349","","","","Inactive","2008-04-23 21:00:00.0"
"87","Anne Pettay","Financial Representative","Guardian","anne_pettay@glic.com","(858)755-2910","8587552921","www.glic.com","11238 El Camino Real, Ste. 200, San Diego, CA 92130","Active","2008-09-08 21:00:00.0"
"89","Terry Farruk","Employee Leasing","SuperiorROI","terry@superiorroi.com","(858)436-1180","","","","Inactive","2008-10-15 21:00:00.0"
"90","Chris Wilson","Graphic Artist","Endoxiart","cw@endoxiart.com","(858)768-1281","","","","Active","2008-11-19 21:00:00.0"
"91","Jennifer Tam","Optometrist","","","","","","","InterChapter","2008-12-03 21:00:00.0"
"92","Ana Meyers","Bookkeeper","","","(619)972-2382","","","","Active","2008-12-10 21:00:00.0"
"93","Shane Haithcock","Search Engine Optimization","Alliance Optimization","shane@allianceoptimization.com","(619)723-6060","","","","Inactive","2008-12-10 21:00:00.0"
"94","Erin Fong","Hair Stylist","","","(805)901-9161","","","","Active","2007-12-31 21:00:00.0"
"95","Paul Devincezo","Florist","","","(619)540-1535","","","","Inactive","2007-12-31 21:00:00.0"
"103","Kathy Deneen","Event Planner","","","(858)848-4843","","","","Inactive","2009-02-05 13:54:02.0"
"106","Seth Obyrne","Real Estate Residential","","","(858)869-3940","","","","Active","2009-03-02 06:07:56.0"
"107","Derek Wattenbarger","General Contractor","","surfside.dev@gmail.com","(619)944-4549","","","","Inactive","2009-03-01 21:00:00.0"
"108","Andreas Koessler","Photographer","","","(760)969-3239","","","","Inactive","2009-03-26 15:00:35.0"
"109","Jamelin Johnson","Loan Officer","","","(619)840-7032","","","","Active","2009-03-01 21:00:00.0"
"110","Matin Atrushi","Financial Advisor","","","(858)875-5832","","","","Active","2009-03-05 18:22:07.0"
"111","Mike Deibler","Personal Trainer","","","(858)752-4561","","","","Active","2009-03-05 18:27:09.0"
"112","Brian Geffen","Printer","Duds By Dudes","dudsbydudes@gmail.com","(858)442-5613","","www.dudsbydudes.com","","Inactive","2009-04-20 01:19:27.0"
"118","Dave Connors","Caterer","Pasta Bravo","dave@pastabravo.com","","","","","InterChapter","2009-04-27 08:58:57.0"
"119","Tina Pavez","Life Coach","Success Coach","unknown","","","","","InterChapter","2009-04-27 09:15:11.0"
"120","John McConnell","Pschologist","John McConnell Ph.D","drmcconnell@cox.net","(619)993-3397","","","","Active","2009-04-26 21:00:00.0"
"121","Florence Casiez","Massage Therapist","Wellness Flow","wellnessflow@gmail.com","","","","","Active","2009-04-30 21:00:00.0"
"122","John Scott","Printer","Caboodle Cartridges","","","","","","InterChapter","2009-06-29 21:00:00.0"
"123","Cindy Freeland","Attorney Labor & Employment","Schor & Freeland","","","","","","Active","2009-06-30 21:00:00.0"
"124","Michael Calabrese","Painter","Michael's Painting","","","","","","Active","2009-06-30 21:00:00.0"
"125","Ryan Clardy","Payroll Services","","","","","","","Active","2009-07-10 16:31:04.0"
 